### PR TITLE
add prettier with solidity plugin to format the code

### PR DESCRIPTION
**Pull request by DanielSintimbrean**
### Changes
* Add and configured prettier. 
* Add script to format `npm run format`
* Fix config.ts
* Add eslint

Now we have type inference when do `ethers.getContractFactory` 
image

</br>
and autocompletion

image

### CODE REVIEW

**Comment by Granahh** [comment]
I agree, maybe we can add this linter as well.
https://typescript-eslint.io/

It would be great if we add a github workflow to review all these rules

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+artifacts/*
+cache/*
+typechain-types/*
```
Comment (Granahh) [inline .prettierignore:3]: Add new line at the end of file
```suggestion
typechain-types/*

```

**Review by Granahh — CHANGES_REQUESTED** [review]

**Comment by Granahh** [comment]
Modify the PR to merge with the develop branch.

**Comment by DanielSintimbrean** [comment]
> I agree, maybe we can add this linter as well. https://typescript-eslint.io/
> 
> It would be great if we add a github workflow to review all these rules

I'm fine with that, i can do it

**Pull request merged**